prime-router/src/main/kotlin/serializers/Hl7Serializer.kt

```diff
@@ -2119,6 +2124,42 @@ class Hl7Serializer(
         val ncesLookupTable = lazy {
             Metadata.getInstance().findLookupTable("nces_id") ?: error("Unable to find the NCES ID lookup table.")
         }
+
+        /**
+         * Walks all the NTE segments and concatenates the values into a single NTE
+         * segment we will write out later on when we deserialize. This is very hackish
+         * and should not be considered a good or permanent solution
+         */
+        fun decodeNTESegments(message: ca.uhn.hl7v2.model.Message): String {
+            // cast the message to an ORU_R01, and if it's not that type of
+            // message, just return an empty string
+            val oruR01: ORU_R01 = (message as? ORU_R01) ?: return ""
+            val sb = StringBuilder()
+
+            // walk all the patient results, and the notes that are listed in there, appending to the stringbuilder
+            oruR01.patienT_RESULTAll.forEach { patientResult ->
+                patientResult.patient.nteAll.forEach { patientNote ->
+                    sb.append(patientNote.comment.joinToString(" ") { it.value })
+                    sb.append(" ")
+                }
+                // inside each patient record is the order observation, which also has notes
```

Comment:
These notes are sometimes very specific to observations. I am not sure if bundling them up like this would work to generate HL7 back.  For example:
1. Each HL7 message can have patient notes.  Those are for the patient regardless of any observations or orders.
2. In an HL7 you can have multiple orders and each order can have notes. Do you always get only one order (one OBR segment)?
3. Within those orders, you can have multiple observations and the notes will belong to one of those observations.  Do you always get only one OBX segment?

If you get only one of all those segments then you may be fine, but you will not be able to position the notes in the correct location of the HL7 if you bundle them all.

Reply:
@carlosfelix2 I know that they are location specific, but as you also know, our current model for structuring messages doesn't really support tying messages to a specific segment, at least not without some rework, and I don't think that the work is justified given how close you all are with FHIR. At this point, this is a "close enough" solution that ensures the notes carry through.